applications/dashboard/settings/class.hooks.php

```diff
@@ -663,7 +663,13 @@ public function gdn_dispatcher_appStartup_handler($sender) {
                 redirectTo($url);
             }
         }
+    }
+
+    public function gdn_auth_startAuthenticator_handler($sender) {
         $this->checkAccessToken();
+        if (Gdn::session()->isValid()) {
+            $sender->authHandled = true;
```

Comment:
This event property kinda seems redundant. Couldn't I check the result of `Gdn::session()->isValid()`, here and in other areas, to see if authentication had already been handled? It's a very lightweight method.

Reply:
Maybe I misunderstood this but when I proposed using the isValid method of the session, @tburry wanted a way to invalidate a previous auth if we need it in the future. He can probably provide deeper info.

Once this is settled I will definitely add a doc bloc for this.

Reply:
I'm going to go back on what I said and agree with @initvector to make things simpler. We already have an improved way of doing this kind of thing in the future so we can just use `Gdn::session()->isValid()` on this for now. Sorry for the doublespeak.